### ISSUE TITLE
Weird mismatch with language-service and @angular/compiler-cli errors

### ISSUE THREAD

**Issue by jpike88** [angular/vscode-ng-language-service#380]
@angular/language-service with latest vscode extension brings up an errors that ```$(npm bin)/ngc -p client/src/tsconfig.json --noEmit``` doesn't bring up, and vice versa. I thought @angular/language-service uses the same error detection logic as @angular/compiler-cli?

**Here is an example of an error that gets shown in the VSCode extension, but not in ngc.**

```
components/app/admin-panel/admin-users.component.ts.AdminUsersComponent.html(3,3): Property 'loading' does not exist on type 'AdminUsersComponent'.
```

```
@Component({
	template: `
			<custom-table *ngIf="!loading">
	                </custom-table>
	`
})
export class AdminUsersComponent {

}
```

^ this is the code that threw the error. This is good. But the VSCode extension does not pick this up.

**Here is an example of an error that gets shown in the VSCode extension, but not in ngc.**

<img width="998" alt="Screen Shot 2019-09-04 at 5 48 41 PM" src="https://user-images.githubusercontent.com/9585787/64248938-aaad9200-cf3c-11e9-878a-56206aa2cd5c.png">

I'm using "@angular/language-service": "^9.0.0-next.4". I'm using VSCode extension version v0.900.0-next.0

**Comment by kyliau** [comment]
This is not due to the latest extension. It has been the way since its inception (type checking works slightly differently under language service vs the compiler), and this will be addressed later this year as part of the rewrite of the language service to use Ivy.

**Comment by jpike88** [comment]
- Does this also explain https://github.com/angular/vscode-ng-language-service/issues/381?
- Also, if the language service is 'behind' ngc (which is what I'm understanding it as), why didn't ngc pick up an error that the language service did?

**Comment by ayazhafiz** [comment]
> * Does this also explain #381?

Yes

> * Also, if the language service is 'behind' ngc (which is what I'm understanding it as), why didn't ngc pick up an error that the language service did?

The language service is not behind `ngc` (the legacy compiler); it currently uses `ngc`, but as Keen mentioned, the language service performs some analyses a bit differently than how `ngc` does. Because Ivy, or `ngtsc` (the new compiler) is expected to land in Angular 9, the language service is not prioritizing fixing bugs in the version written with `ngc`. Also as Keen mentioned, this will be fixed when the language service is re-written to use Ivy 🙂.

**Comment by Maximaximum** [comment]
I have found a similar issue and had to spend many hours debugging in order to realize that the vscode ng service extension reports different errors compared to what angular Ivy compiler does. This feels really weird.

I really hope that this issue will be addressed by the extension team soon.

**Comment by ayazhafiz** [comment]
@Maximaximum please see https://github.com/angular/vscode-ng-language-service/issues/335#issuecomment-615008175

**Comment by kyliau** [comment]
This has been fixed by the new Ivy-native language service, released in [v11.1.0](https://github.com/angular/vscode-ng-language-service/releases/tag/v11.1.0).
